src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java

```diff
@@ -136,87 +137,90 @@ SecurityAnalysisReport runSync(String workingVariantId, SecurityAnalysisParamete
         // CosPhi for DC power to current conversion
         OpenLoadFlowParameters parametersExt = OpenLoadFlowParameters.get(securityAnalysisParameters.getLoadFlowParameters());
         DcSecurityAnalysisContext context = new DcSecurityAnalysisContext(securityAnalysisParameters, contingencies, detector, parametersExt.getDcPowerFactor());
-        for (Branch<?> b : network.getBranches()) {
-            context.getFactors().add(new SensitivityFactor(SensitivityFunctionType.BRANCH_ACTIVE_POWER_1, b.getId(), SensitivityVariableType.INJECTION_ACTIVE_POWER,
-                    variableId, false, contingencyContext));
-        }
-        SensitivityAnalysisResult result = new SensitivityAnalysis.Runner(sensitivityAnalysisProvider)
-                .run(network, workingVariantId, context.getFactors(), context.getContingencies(), variableSets, sensitivityAnalysisParameters, computationManager, Reporter.NO_OP);
+        SensitivityFactorReader factorReader = handler -> {
+            for (Branch<?> b : network.getBranches()) {
+                SensitivityFactor f = new SensitivityFactor(SensitivityFunctionType.BRANCH_ACTIVE_POWER_1, b.getId(), SensitivityVariableType.INJECTION_ACTIVE_POWER,
+                        variableId, false, contingencyContext);
+                context.getFactors().add(f);
```

Comment:
We don't really need to save factors because we just get the branch id when writing values and we could directly using the factorIndex go into the branches array, as there is as many factors than branches.

Reply:
Anyway it won't change anything to performance